crypto/CMakeLists.txt

```diff
@@ -111,14 +111,7 @@ if (CONFIG_NRF_CC3XX_PLATFORM)
   #
   zephyr_sources(${NRF_CC3XX_PLATFORM_BASE}/src/nrf_cc3xx_platform_abort_zephyr.c)
   zephyr_sources(${NRF_CC3XX_PLATFORM_BASE}/src/nrf_cc3xx_platform_mutex_zephyr.c)
-endif()
 
-if (CONFIG_CC3XX_BACKEND)
-  if(CONFIG_SOC_NRF5340_CPUAPP)
-    set (CC3XX_ARCH cc312)
-  else()
-    set (CC3XX_ARCH cc310)
-  endif()
   set(NRF_CC3XX_BASE ${CMAKE_CURRENT_SOURCE_DIR}/nrf_${CC3XX_ARCH}_mbedcrypto)
```

Comment:
when removing `CC3XX_ARCH` variable, above this line will become:
```
nrf__mbedcrypto
```
which i'm confident will break, as there are two `_` after each other.

Reply:
I didn't remove the variable CC3XX_ARCH. I merged two if statements to one and both of them set these variables at the top.  https://github.com/nrfconnect/sdk-nrfxlib/blob/71476ebb6101351a27559ea07cddd8787a9ff3fc/crypto/CMakeLists.txt#L80-L84

Reply:
right, I see that, because you also remove the guard: `if (CONFIG_CC3XX_BACKEND)`

But that means that the code part that used to be guarded by `if (CONFIG_CC3XX_BACKEND)` is now guarded by `if (CONFIG_NRF_CC3XX_PLATFORM)` instead, but I can have:
```
CONFIG_NRF_CC3XX_PLATFORM=y
CONFIG_CC3XX_BACKEND=n
```
so what is the reason that the code which used to be guarded by `if (CONFIG_CC3XX_BACKEND)` is now being included when `CONFIG_CC3XX_BACKEND=n` ?

Reply:
You are correct, the previous version could have linked more libraries than needed when CONFIG_CC3XX_BACKEND=n. With the current solution we only linking these libraries only when needed.